pkg/router/template/router_test.go

```diff
@@ -1094,6 +1094,51 @@ func TestCalculateServiceWeights(t *testing.T) {
 			},
 			expectedWeights: map[ServiceUnitKey]int32{},
 		},
+		{
+			name:      "a single service with one endpoint with weight 0",
+			routePort: "8080",
+			serviceUnits: map[ServiceUnitKey][]Endpoint{
+				suKey1: {ep1, ep2},
+			},
+			serviceWeights: map[ServiceUnitKey]int32{
+				suKey1: 0,
+			},
+			expectedWeights: map[ServiceUnitKey]int32{
+				suKey1: 0,
+			},
+		},
+		{
+			name:      "services with multiple units including one shut",
```

Comment:
*nit* consistent naming pattern with other unit tests and not sure what a `unit` here is.

```suggestion
			name:      "services with one of weight 0",
```